src/sas/qtgui/Plotting/Plotter2D.py

```diff
@@ -704,19 +703,18 @@ def showPlot(self, data, qx_data, qy_data, xmin, xmax, ymin, ymax,
             data_x, data_y = numpy.meshgrid(self.data0.x_bins[0:-1],
                                             self.data0.y_bins[0:-1])
 
-            ax = Axes3D(self.figure)
+            self.ax = self.figure.add_subplot(111, projection='3d')
 
             # Disable rotation for large sets.
             # TODO: Define "large" for a dataset
             SET_TOO_LARGE = 500
             if len(data_x) > SET_TOO_LARGE:
```

Comment:
I guess in majority (or even all cases) data_x is same/similar to data_y but wondering if this conditions shouldn't be generalized for data.size